File: leetcode Solns/Daily Challenges/paranthesis_score.py
class Solution:
    def scoreOfParentheses(self, S: str) -> int:
        score,curr = 0,0
        for i in range(len(S)):
            if S[i] == '(':
                curr+=1
            else:
                curr-=1
                if S[i-1] == '(':
                    score += 1<<curr
        return score

File: leetcode Solns/Daily Challenges/paranthesis_score.java
class Solution {
    public int scoreOfParentheses(String s) {
        int score = 0, curr=0;
        for(int i=0;i<s.length(); i++){
            if(s.charAt(i) == '(')
                curr++;
            else{
                curr--;
                if(s.charAt(i-1) == '(')
                    score += 1<<curr;
            }
        }
        return score;
    }
}